continuous_delivery/continous_integration_continous_delivery.md

```diff
@@ -0,0 +1,43 @@
+
+### Building
+1. Style/lint checks. Ensure consistency in coding styles and convention makes code easier to read and understand.
+2. Static analysis tools. Scans code for quality and potential security risks.
+3. Dependency auditing. Checks third party package for vulnerabilities and updates.
+4. Compilation (for compiled code). Catches syntax/compliation errors.
+
+### Testing
+1. Run unit tests. Code changes include new unit tests and modifying existing tests to ensure functionality is working as expected. 
+2. Enforce test Coverage. Ensures code base meets testing coverage threshold.
+3. Deploy code to testing environment. Ideally an ephemeral environment that will be used for functional testing.  Otherwise use a blocking environment that serializes pull request jobs during the functional tests.
+4. Run database migrations. Scaffolding database and running migrations to build a consistent database that mirrors those in production. 
+5. Seed any new application database data. Also seed any test data required for functional tests.
+6. Run functional/integration tests. These tests are written from the perspective of end user/application feature perspective. A typical example uses browser drivers that mimic user interactions with the application, in a repeatable deterministic script. 
+
+### Code Review
+1. Peer code review and approval. Peer review is ready once all automated checks passes.  This ensures that the changes meet the minimum validation threshold before review begins. Any new code changes during the review will trigger the automation pipeline again.
+2. Merge code change. Once the pull request is approved, the author merges the changes.
+
+## Continuous Delivery
+
+Merging code changes to the development branch triggers automation to deploy the code to a staging environment.
+
+1. Deploy code to qa/staging environment. This environment is a long-lived environment for feature acceptance.  
+2. Run database migrations.  Run any wew migrations. 
+3. Populate database seed data.  Populate any new seed data.
+4. Feature Acceptance on staging environment.
+
+## Production Deployment
+
+Production deployment strategy should include handling rollback and recovery in case of any issues that might happen during deployment. Using release tagging with a version control like Git can help reverting the application to a previously known working state. Combined with regular backups of application data and configuration, will enable fast rollback in case of failure. The follow are steps recommended when deploying to production.
+
+1. Create a release tag in source control. This will create a snapshot of the codebase for the code that will be deployed to production.
+2. Manual approval for merging of the release tag to the source control. This acts as the final trigger for deployment to production.
```

Comment:
There are inconsistencies in the terminology used throughout the guide. At times, "source control" is referenced without an article, while in other instances, it's preceded by "the."

```suggestion
2. Manual approval for merging of the release tag to source control. This acts as the final trigger for deployment to production.
```

There's also some inconsistency in sentence structure for these steps...
For instance, in Step 1, `Create a release tag in source control`, the language resembles an instructional command. Conversely, in Step 2, `Manual approval for merging of the release tag to source control`, the phrasing lacks coherence as it appears more like a topic than a complete sentence. Achieving consistency in both terminology and sentence structure across all steps would enhance clarity and readability.  

**I think this applies to the other sections of this doc as well.**

Reply:
Let me do another path through this to try to fix up the sentence structure.  This is definitely a product of my "bullet point" draft -> turning it into a "guide" approach to writing this up.